packages/client/src/modals/TransactionSuccessModal.js

```diff
@@ -0,0 +1,131 @@
+import React, { useContext, useEffect, useState } from "react";
+import { Web3Context } from "contexts/Web3";
+import { useClipboard, useContacts } from "../hooks";
+import { Copy, ArrowUp, OpenNewTab } from "../components/Svg";
+import { getFlowscanUrlForTransaction } from "utils";
+import { ASSET_TYPES } from "../constants/enums";
+import { COIN_TYPE_TO_META, CONTRACT_NAME_TO_COIN_TYPE } from "../constants/maps";
+
+const TransactionSuccessModal = ({ actionData, actionType, txID, onClose, safeName, safeAddress }) => {
+    const { getNFTReference } = useContext(Web3Context);
+    const clipboard = useClipboard();
+    const [image, setImage] = useState();
+    const { contacts } = useContacts(safeAddress);
+
+    const { recipient, nftId, collectionId, vaultId, tokenAmount } = actionData;
+    const { displayName, icon } = getTokenMeta(vaultId) || {};
+    const { name: imageName, imageURI } = image || {};
+
+    const NFTName = collectionId?.split(".")[2];
+
+    useEffect(() => {
+        if (actionType === ASSET_TYPES.NFT) {
+            const getImageURL = async () => {
+                const result = await getNFTReference(recipient, nftId);
```

Comment:
`getNFTReference` is currently hardcoded with ZeedzINO, we should make it generic.